runtime/include/spinlock_wrapper.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef SPINLOCK_WRAPPER_HPP
+#define SPINLOCK_WRAPPER_HPP
+
+#include <new>
+#include <pthread.h>
+#include <atomic>
+
+// Include the custom spinlock implementation
+#include "spinlock.hpp"
+
+// Define the custom spinlock type and functions only if the standard library is not available
+#if !defined(_POSIX_SPIN_LOCKS) || _POSIX_SPIN_LOCKS <= 0
+typedef Spinlock pthread_spinlock_t; // Define custom spinlock type
+
+// Non-volatile version
+inline int pthread_spin_init(pthread_spinlock_t* lock, int pshared) {
+    (void)pshared; // suppress unused parameter warning
```

Comment:
What about using `[[maybe_unused]]`? 

See https://en.cppreference.com/w/cpp/language/attributes/maybe_unused

Reply:
okay, will incorporate the above change